### PR TITLE
Homework 2

### PR DESCRIPTION
**Pull request by halltim91**
Added my solutions for homework 2.

### CODE REVIEW

File: hw2/05-form.html
Comment (NathanMoes) [inline hw2/05-form.html:1]: image
I do not see any options for the drop down. But it does not seem to be spec'ed in the hw2 so. Idk. Just a thing I noticed I suppose. I believe I put some options in there for registered and not etc.

File: hw2/05-form.js
Comment (NathanMoes) [inline hw2/05-form.js:1]: I personally feel a formatted template litteral is better here.
Just a nitpick

File: hw2/04-background.js [hunk 1]
```diff
@@ -1 +1,45 @@
...
+    }
+}
+
+function setBackgroundColor(){
+    const r = Math.floor(Math.random() * 255);
+    const g = Math.floor(Math.random() * 255);
+    const b = Math.floor(Math.random() * 255);
+    const a =  .5 + (Math.random() * .4);
+    const color = 'rgba(' + r + ',' + g + ',' + b + ',' + a + ')';
+    body.style.backgroundColor = color;
```
Comment (NathanMoes) [inline hw2/04-background.js:32]: A template litteral here would be far more reabale 
something like color = `rgba(${value}...)`

File: hw2/03-count.js [hunk 2]
```diff
@@ -1,3 +1,16 @@
-// Add your code here
+const textInput = document.getElementById('input');
+const text = document.getElementById('text');
+const origText = text.innerHTML;
 
-input.addEventListener('keydown', handleKeyDown);
+textInput.addEventListener('keydown', handleKeyDown);
+
+function handleKeyDown(event) {
```
Comment (NathanMoes) [inline hw2/03-count.js:7]: Also nitpick. I feel the thing should search the whole thing while entering each key, and not wait till enter key. But also not given in the spec so... Just a sugestion. Personally like the continual search more. But it does not matter

Comment (NathanMoes) [reply hw2/03-count.js:7]: Still a good impl here. Only exec if we hit an enter key, works well. Good job!

**Review by NathanMoes — APPROVED** [review]

**Comment by shrutivd** [comment]
Overall the code looks simple and readable. I would suggest the same for the string literal in rgba to use rgba${value}.. as Nathan did. Everything else looks good to me.

**Review by shrutivd — APPROVED** [review]

File: hw2/05-form.html
Comment (halltim91) [reply hw2/05-form.html:1]: Yeah, I saw you did and I can't believe it never occurred to me what the options would be for this lol

File: hw2/04-background.js (hunk 1 above)
Comment (halltim91) [reply hw2/04-background.js:32]: I agree. Forgot about those.

**Pull request merged**